torch/package/package_exporter.py

```diff
@@ -210,6 +212,12 @@ def get_unique_id(self) -> str:
         self._unique_id += 1
         return ret
 
+    def get_storage_id(self) -> int:
```

Comment:
(not a change requested on this PR, but we should file as a followup) couple of thoughts on this API.

1. This is probably mergeable with `get_unique_id`. 
- I think it's potentially confusing that `get_unique_id` returns an id that could potentially collide with an id returned by `get_storage_id`, since we kind of sort of promised that it would be "unique", and people may interpret that as "unique relative to the entire package" as opposed to "unique relative to calls to get_unique_id". 
- Zooming out a bit: we want a generic way for users using `__reduce_package__` to grab a unique id and not worry about colliding with anyone else, so long as they use the name in a unique way (like `<id>.storage`). So in that sense `get_storage_id` and `get_unique_id` should be the same.

2. I think the API for `get_unique_id`/`get_storage_id` is not quite right. Right now, we are getting a single id, then relying on an implementation detail (that the ids are ints and the way we generate them is incrementally ascending) to allocate as many ids as we want, then we rely on another implementation detail (`_next_storage_id` existing) and set it directly from an external source (`ScriptModule.__reduce_package__`). I think this is probably too much cooperation for a core serialization API. 

An alternative could be to change the API to be "get me N unique ids". That nicely encapsulates the implementation, but comes at the cost that you'll need to know how many ids you need up front.